pkg/diff/merge.go

```diff
@@ -41,11 +41,10 @@ func (r RowDatas) Less(i, j int) bool {
 		data1 = r.Rows[i].Data[col.Name.O]
 		data2 = r.Rows[j].Data[col.Name.O]
 		if needQuotes(col.FieldType) {
-			if string(data1) > string(data2) {
-				return false
-			} else if string(data1) < string(data2) {
-				return true
-			} else {
+			strData1 := string(data1)
+			strData2 := string(data2)
+
+			if strData1 == strData2 {
```

Comment:
`len(strData1) == len(strData2) && ..`

Reply:
@lonng  I disagree with the `len(a) == len(b) && a == b` check because there is no algorithmic advantage at all. Firstly, `a == b` already does the length check (see golang/go#1161).

Now, if `len(a) == len(b)`, then `a == b` will do the O(n) comparison anyway. If `len(a) != len(b)` then it will still perform the check `strData1 > strData2` which is O(n) again.

According to the comment [in `strings.Compare`](https://golang.org/src/strings/compare.go), the pattern 

```go
if a == b {
    return foo;
}
if a < b {
    return bar;
}
return bar;
```

will always be optimized into a [single](https://github.com/golang/go/blob/master/src/internal/bytealg/compare_generic.go#L39-L60) O(n) string comparison 

> NOTE(rsc): This function does NOT call the runtime cmpstring function,
> because we do not want to provide any performance justification for
> using strings.Compare. Basically no one should use strings.Compare.
> As the comment above says, it is here only for symmetry with package bytes.
> **If performance is important, the compiler should be changed to recognize
> the pattern so that all code doing three-way comparisons**, not just code
> using strings.Compare, can benefit.

IMO all those `len(a) == len(b) &&` check should be removed unless there's a performance test showing it is indeed faster.

Reply:
OK